zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/SelectTagKeys.java

```diff
@@ -0,0 +1,82 @@
+package zipkin2.storage.cassandra;
+
+/*
+ * Copyright 2015-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.ResultSetFuture;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+import zipkin2.Call;
+import zipkin2.storage.cassandra.internal.call.AccumulateAllResults;
+import zipkin2.storage.cassandra.internal.call.ResultSetFutureCall;
+
+import static zipkin2.storage.cassandra.Schema.TABLE_TAGS;
+
+public class SelectTagKeys extends ResultSetFutureCall {
+
+  static class Factory {
+    final Session session;
+    final PreparedStatement preparedStatement;
+    final AccumulateTagsAllResults accumulateTagsIntoSet =
+      new AccumulateTagsAllResults();
+
+    Factory(Session session) {
+      this.session = session;
+      this.preparedStatement =
+        session.prepare(QueryBuilder.select("key").distinct().from(TABLE_TAGS));
```

Comment:
I don't know what the cardinality of tags will be? But this won't scale.

But, for example, if there's 1 million tags in a Zipkin storage this full-table scan is going to be painful (if not timeout).

Reply:
Right. I would do a progressive query (by pagination but only simple one, no ordering and deifnirively not with `offset` but with `after`.

Reply:
If we want to have a whitelist of tags to index and not just index all tags you might get away without doing this at all. And just return the list of whitelisted tags.

Reply:
Added the whitelist implementation. Let me know your thoughts.